### ISSUE TITLE
Removal of whitespace in f-string when used with python 3.12 (not with python 3.11)

### ISSUE THREAD

**Issue by corneliusroemer** [snakemake/snakefmt#227]
When running snakefmt through pre-commit I noticed an interesting edit.

In this rule:

```py
rule subsample:
    input:
        metadata=build_dir + "/{build_name}/good_metadata.tsv",
    params:
        exclude=lambda w: (
            f"--exclude-where {' '.join([f'lineage={l}' for l in config['subsample'][w.sample]['exclude_lineages']])}"
            if "exclude_lineages" in config["subsample"][w.sample]
            else ""
        ),
```

in the following line:

from:

```
            f"--exclude-where {' '.join([f'lineage={l}' for l in config['subsample'][w.sample]['exclude_lineages']])}" 
                                                       ^ this space
```

to:

```
            f"--exclude-where {' '.join([f'lineage={l}'for l in config['subsample'][w.sample]['exclude_lineages']])}"
                                                       ^ is removed by v0.10.1 with python 3.12 only
```

Note: the space after `'` in `' for l in config` was removed.

I'm pretty sure pre-commit uses python3.12 on my system by default. I managed to work around this by telling pre-commit to use python3.11 (which is what I use in VScode's snakefmt).

This is another python3.12-only bug/change, similar to: https://github.com/snakemake/snakefmt/issues/207 and https://github.com/snakemake/snakefmt/issues/220

Workaround for all python3.12-only issues when using pre-commit:

```
  - repo: https://github.com/snakemake/snakefmt
    rev: v0.10.1
    hooks:
      - id: snakefmt
        language_version: python3.11  # <--- Add this line
```

**Comment by mbhall88** [comment]
Yeah, this is very tightly linked to #220. Hopefully @bricoletc or I can get around to fixing this soon. I tried recently but couldn't quite get it sorted and then had to go back to other projects.

Feel free to tackle it also @corneliusroemer if you feel up to it

**Comment by mbhall88** [comment]
this should be fixed in v0.10.2
